### ISSUE TITLE
Wie die Tabellen "bs_users u. bs_users_meta" auf eine Neuinstallation übertragen

### ISSUE THREAD

**Issue by tennstar1** [tkrebs/ep3-bs#540]
Hallo,
ich musste leider das System neu installieren und würde nun gerne die beiden Tabellen "bs_users u. bs_users_meta" aus dem alten System in das neue übertragen um ein generelles "Neu" anmelden der Teilnehmer zu verhindern.

Das Importieren der beiden Tabellen wird abgebrochen mit der Fehlermeldung, sie wären schon vorhanden.

Ein löschen der beiden vorhandenen wird von der Datenbank abgelehnt und ist nicht möglich.

Hat bitte jemand einen Lösungsansatz wie das zu lösen ist für einen Tipp wäre ich sehr dankbar! 
Leider bin ich kein Datenbank Freak!

Gruß tennstar1

**Comment by cbits68** [comment]
hast Du die beiden Tabellen "truncated" bevor Du sie importiert hast?
truncate leert die Tabelle vollständig!
Wenn Du neu installiert hast ist ja logischerweise mindestens der Admin User schon  angelegt. Damit kannst Du dann natürlich nicht importieren, weil ja schon Einträge vorhanden sind.
Löschen geht ja auch nicht, weil ja noch Einträge vorhanden sind ...
Gruß CBITS

**Comment by tennstar1** [comment]
Danke, werde mal eine Datenbank Sicherung anlegen und versuchen dies umzusetzen!

Danke für den Tipp!

**Comment by tennstar1** [comment]
cbit68
So, hab die Tabellen gelehrt und die beiden importiert, leider kommt wieder die Fehlermeldung
"#1050 - Tabelle 'bs_users' bereits vorhanden".

Was ist verkehrt?

**Comment by cbits68** [comment]
OK, kannst Du die Tabelle denn jetzt löschen?

**Comment by cbits68** [comment]
Es ist halt etwas komplexer, da Du ja neben Tabellen und Daten in einer Datenbank auch Verknüpfungen zwischen Tabellen hast. Ich gehe daher eher den Weg die alte Datenbank zu bereinigen und dann komplett zu übernehmen, anstatt einzelne Tabellen zu verändern oder komplett zu löschen.

**Comment by tennstar1** [comment]
Konnte sie löschen, aber die beiden anderen nicht importieren, wieder eine Fehlermeldung.

Was meist Du mit bereinigen der alten Datenbank?

**Comment by cbits68** [comment]
Damit meine ich, dass ich in der alten Datenbank alle Buchungen bookings, reservations und events gelöscht habe. Dann habe ich die Datenbank exportiert, eine neue (leere) DB angelegt und dieb Export der alten DB in die neue DB importiert.
Danach habe ich in der ep3 Installation die Zugangsdaten der neuen DB eingetragen und hatte  so die komplette Konfig wieder verfügbar, inkl Nutzer, Passörter, Plätze, ... aber eben ohne die alten Buchungen.

**Comment by tennstar1** [comment]
Danke, werd´s jetzt mal ein bisschen sacken lassen und morgen mal drüber gehen!

Nochmals Danke für Deine Hilfe

Schönen Abend tennstar1

**Comment by mjmpeace** [comment]
Ich höre mal zu - weil ich gerade das Gleiche vorhabe. Allerdings will ich in eine neue Installation die komplette alte Datenbank einbinden, weil ich ja alles übernehmen will. Wie mache ich das denn? Einfach in der Installation die Datenbank Einträge ändern? Geht das ?

**Comment by cbits68** [comment]
ja, warum nicht? Verbinde doch mal die neue Version mit der alten Datenbank .... soweit mir bekannt wurden an den Tabellen zwischen 1.6.3  und 1.7.0 ja nichts verändert.

**Comment by tennstar1** [comment]
cbits68

Dein Hinweis war Gold wert! Hab´s so gemacht wie Du beschrieben hast und es läuft wieder und alle Teilnehmer sind wieder vorhanden!

Danke nochmals für den Tipp

Gruß tennstar1

**Comment by mjmpeace** [comment]
Super dann gehe ich da auch ran. Danke

**Comment by mjmpeace** [comment]
@tennstar1 wie hast du das mit den Verzeichnissen gelöst? Ich habe die neue Installation in ein Verzeichnis Test gemacht. Das soll eigentlich nicht so bleiben, sondern wieder "halle" heißen. Umbenennen funktioniert aber nicht einfach. Da kommt Fehler 500.
Was ist zu tun?

**Comment by tennstar1** [comment]
@mjmpeace
nachdem ich die neue Version installiert und alle Einstellungen und Änderungen aus der alten übernommen hatte und diese auch funktioniert hat, habe ich die alte Datenbank in Xammp geöffnet und alle Einträge, wie von cbits68 geschildert entfernt. Die installierte neue Datenbank habe ich nun zunächst mal gesichert, gelöscht und dann die bereinigte aus Xammp importiert. Alles läuft dann wieder und die Anmeldungen im System sind vorhanden und funktionieren. Ob diese Methode ideal ist kann ich nicht sagen, aber sie hat funktioniert!

**Comment by cbits68** [comment]
Noch als Ergänzung. Ich mache die Übung (Löschen der alten Buchungen vom Vorjahr) jedes Jahr vor Saisonbeginn, um die Datenbank möglichst schmal zu halten und auch den DSGVO Vorgaben genüge zu tun.
Dafür kann man auch direkt auf die Datenbank mittels phpMyAdmin zugreifen, was die meisten Provider ja auch anbieten.
Dabei lösche ich alle Inhalte in den Tabellen für bookings, reservations & events:
![image](https://user-images.githubusercontent.com/60112941/155485138-75825114-52d8-4421-8f1c-f2f99244e772.png)

Alles Andere wie Einstellungen, Benutzer, Berechtigungen und Passwörter bleibt somit erhalten und nur die alten Buchungen sind somit entfernt.

**Comment by mjmpeace** [comment]
Ich habe den einfachen Weg gewählt. Neue Installation mit neuer Datenbank. Test. Dann die alte Datenbank eingebunden in der local.php. Verzeichnis umbenannt wie die alte Installation. Altes Verzeichnis in old  umbenannt, Cache bereinigt. Dann lief alles wie vorher und durch dem neuen Server werden jetzt auch wieder Mails versandt an Adressen, die auf dem alten Server mit tls connection failed nicht versendet werden konnten. 
Nun muss ich in Ruhe noch das Outdoor System reparieren.
